File: lab/explorer/cpp/greet/greet.cpp
#include <iostream>  // For std::cout, std::cin, std::endl (input/output operations)
#include <string>    // For std::string data type and string operations

std::string greet(const std::string& name)
{
	return "Hello, " + name + "!";
}

int main()
{
	std::string name;

	std::cout << "What's your name? ";
	std::getline(std::cin, name);

	std::string message = greet(name);
	std::cout << message << std::endl;

	return 0;
}

File: lab/explorer/rust/greet/greet.rs
use std::io; // brings input/output into scope so we can use io::stdin() instead of std::io::stdin()

fn main() {
    println!("What's your name?");

    let mut name = String::new();
    io::stdin()
        .read_line(&mut name)
        .expect("Failed to read line");

    // remove the trailing newline
    let name = name.trim();

    println!("Hello, {}!", name);
}
